common/src/main/java/com/alibaba/nacos/common/remote/client/RpcClient.java

```diff
@@ -913,25 +904,17 @@ protected ServerInfo currentRpcServer() {
      */
     @SuppressWarnings("PMD.UndefineMagicConstantRule")
     private ServerInfo resolveServerInfo(String serverAddress) {
-        String property = System.getProperty("nacos.server.port", "8848");
-        int serverPort = Integer.parseInt(property);
-        ServerInfo serverInfo = null;
-        if (serverAddress.contains(Constants.HTTP_PREFIX)) {
-            String[] split = serverAddress.split(Constants.COLON);
-            String serverIp = split[1].replaceAll("//", "");
-            if (split.length > 2 && StringUtils.isNotBlank(split[2])) {
-                serverPort = Integer.parseInt(split[2]);
-            }
-            serverInfo = new ServerInfo(serverIp, serverPort);
-        } else {
-            String[] split = serverAddress.split(Constants.COLON);
-            String serverIp = split[0];
-            if (split.length > 1 && StringUtils.isNotBlank(split[1])) {
-                serverPort = Integer.parseInt(split[1]);
-            }
-            serverInfo = new ServerInfo(serverIp, serverPort);
+        Matcher matcher = EXCLUDE_PROTOCOL_PATTERN.matcher(serverAddress);
```

Comment:
Why change this logics?

Your Pr is add unit test. Not refactor.

Reply:
already dropped